src/MCPClient/lib/clientScripts/validateFile.py

```diff
@@ -1,91 +1,297 @@
+        try:
+            sip_model = SIP.objects.get(uuid=self.sip_uuid)
+        except (SIP.DoesNotExist, SIP.MultipleObjectsReturned):
+            print('Warning: unable to retrieve SIP model corresponding to SIP'
+                  ' UUID {}'.format(self.sip_uuid), file=sys.stderr)
+            return None
+        else:
+            sip_path = sip_model.currentpath.replace(
+                '%sharedPath%', self.shared_path, 1)
+            logs_dir = os.path.join(sip_path, 'logs')
+            if os.path.isdir(logs_dir):
+                self._sip_logs_dir = logs_dir
+                return logs_dir
+            print('Warning: unable to find a logs/ directory in the SIP'
+                  ' with UUID {}'.format(self.sip_uuid), file=sys.stderr)
+            return None
+
+    @property
+    def sip_pres_val_dir(self):
+        """Return the full path to the directory within the SIP where stdout
+        from perservation derivative validation output should be written to
+        disk.
+        TODO: the leaf dir of the path return is 'implementationChecks' because
+        this functionality was implemented with MediaConch implementation
+        checks in mind. However, if saving command stdout to the logs/ dir of a
+        SIP is needed for validation based on other tools, this directory name
+        may need to be changed in a context-sensitive way.
+        """
+        if self._sip_pres_val_dir:
+            return self._sip_pres_val_dir
+        if self.sip_logs_dir:
+            _sip_pres_val_dir = os.path.join(
+                self.sip_logs_dir, 'implementationChecks')
+            if os.path.isdir(_sip_pres_val_dir):
+                self._sip_pres_val_dir = _sip_pres_val_dir
+            else:
+                try:
+                    os.makedirs(_sip_pres_val_dir)
+                except OSError:
+                    pass
+                else:
+                    self._sip_pres_val_dir = _sip_pres_val_dir
+        return self._sip_pres_val_dir
+
+
+def _get_logger_name(file_type):
+    if file_type == 'original':
+        return "archivematica.mcp.client.validateFile"
+    elif file_type == 'preservation':
+        return "archivematica.mcp.client.validatePreservationDerivative"
     else:
-        return 0
+        return "archivematica.mcp.client.validateAccessDerivative"
+
+
+def _get_shared_path(argv):
+    try:
+        return argv[4]
+    except IndexError:
+        return None
 
-if __name__ == '__main__':
-    logger = get_script_logger("archivematica.mcp.client.validateFile")
 
+def _get_file_type(argv):
+    try:
+        return argv[5]
+    except IndexError:
+        return 'original'
+
+
+if __name__ == '__main__':
     file_path = sys.argv[1]
     file_uuid = sys.argv[2]
     sip_uuid = sys.argv[3]
-    sys.exit(main(file_path, file_uuid, sip_uuid))
+    shared_path = _get_shared_path(sys.argv)
+    file_type = _get_file_type(sys.argv)
+    logger = get_script_logger(_get_logger_name(file_type))
```

Comment:
I think you should use the logger instead of `sys.stderr`. The events are currently [written to disk](https://github.com/artefactual/archivematica/blob/stable/1.6.x/src/archivematicaCommon/lib/custom_handlers.py#L20) but we're planning to change that soon.